test/fixtures/permission-validation/setup.js

```diff
@@ -0,0 +1,482 @@
+  return permissions;
+};
+
+const linkRolesAndUsers = async (roles, users) => {
+  const linkSubsystemRolesAndUsers = async (roles, users) =>
+    Promise.all(
+      Object.keys(roles).map(async role =>
+        pgProvider.linkDetails(
+          'SystemUser',
+          'Roles',
+          users[`${role}User`],
+          roles[role]
+        )
+      )
+    );
+
+  await Promise.all(
+    subsystemNames.map(subsystem =>
+      linkSubsystemRolesAndUsers(roles[subsystem], users[subsystem])
+    )
+  );
+};
+
+const linkActionsAndPermissions = async (actions, permissions) => {
+  const links = [
+    [permissions.S1.BlockedPerson, actions.getPersonName],
+    [permissions.S1.AdminPerson, actions.getPersonName],
+    [permissions.S1.AdminDocument, actions.getDocumentName],
+    [permissions.S1.AdminPersonDocument, actions.getPersonDocumentName],
+  ];
+
+  await Promise.all(
+    links.map(async args =>
+      pgProvider.linkDetails('Permission', 'Actions', ...args)
+    )
+  );
+};
+
+const linkApplicationsAndRoles = async (roles, applications) => {
+  const links = [
+    [roles.S1.Admin, applications.PersonApplication],
+    [roles.S2.Admin, applications.DocumentApplication],
+  ];
+
+  await Promise.all(
+    links.map(async args =>
+      pgProvider.linkDetails('Role', 'Applications', ...args)
+    )
+  );
+};
+
+const startJstpServer = async () => {
+  const user = {
+    login: (connection, userId, callback) => {
+      connection.session.set('userId', userId);
+      callback();
+    },
+  };
+  const remoteProviderApi = createRemoteProviderJstpApi(
+    pgProvider,
+    pgProvider.cursorFactory
+  );
+  const remoteProviderApp = new jstp.Application('remoteProvider', {
+    ...remoteProviderApi,
+    user,
+  });
+  const server = jstp.net.createServer([remoteProviderApp]);
+
+  return new Promise(resolve => server.listen(3000, () => resolve(server)));
+};
+
+const setup = async () => {
+  await preparePgProvider();
+
+  const subsystems = await prepareSubsystems();
+  const roles = await prepareRoles(subsystems);
+  const users = await prepareUsers(subsystems);
+  const [actions, categories, applications] = await getEntitiesIds();
+  const catalogs = await prepareCatalogs(categories);
+  const data = await prepareTestData(categories, catalogs, subsystems);
+  const permissions = await preparePermissions(
+    subsystems,
+    categories,
+    catalogs,
+    roles
+  );
+
+  await linkRolesAndUsers(roles, users);
+  await linkActionsAndPermissions(actions, permissions);
+  await linkApplicationsAndRoles(roles, applications);
+  const server = await startJstpServer();
+
+  const ctx = {};
+
+  ctx.data = data;
+  ctx.roles = roles;
+  ctx.users = users;
+  ctx.server = server;
+  ctx.actions = actions;
+  ctx.catalogs = catalogs;
+  ctx.provider = pgProvider;
+  ctx.subsystems = subsystems;
+  ctx.categories = categories;
+  ctx.permissions = permissions;
+  ctx.applications = applications;
+
+  return ctx;
```

Comment:
You can replace lines 443-457 with a single return statement:
```suggestion
  return {
    data,
    roles,
    users,
    server,
    actions,
    catalogs,
    provider: pgProvider,
    subsystems,
    categories,
    permissions,
    applications,
  };
```